### PR TITLE
Add duct flow tutorial

### PR DESCRIPTION
**Pull request by charleskawczynski**
### Description

This PR adds a boundary condition tutorial. Still very much WIP...

- [ ] Code follows the [style guidelines](https://clima.github.io/ClimateMachine.jl/latest/DevDocs/CodeStyle/) OR N/A.
- [ ] Unit tests are included OR N/A.
- [ ] Code is exercised in an integration test OR N/A.
- [ ] Documentation has been added/updated OR N/A.

### CODE REVIEW

File: tutorials/BalanceLaws/bcs_duct_flow.jl [hunk 1]
```diff
@@ -0,0 +1,828 @@
...
+    xmax,
+    ymax,
+    zmax,
+    param_set,
+    init_state_prognostic!;
+    model = model,
+    xmin = xmin,
+    ymin = ymin,
+    zmin = zmin,
+    numerical_flux_first_order = CentralNumericalFluxFirstOrder(),
```
Comment (charleskawczynski) [inline tutorials/BalanceLaws/bcs_duct_flow.jl:569]: Try RusanovNumericalFlux?
